documentation/src/main/asciidoc/userguide/chapters/query/hql/HQL.adoc

```diff
@@ -1351,7 +1351,83 @@ Applications that aim for database portability should avoid using functions in t
 
 Application developers can also supply their own set of functions.
 This would usually represent either custom SQL functions or aliases for snippets of SQL.
-Such function declarations are made by using the `addSqlFunction()` method of `org.hibernate.cfg.Configuration`.
+Such function declarations are made by using the `addSqlFunction()` method of
+the `org.hibernate.boot.MetadataBuilder` or the legacy `org.hibernate.cfg.Configuration`.
+
+Now, let's assume we have the following `apply_vat` PostgreSQL user-defined function:
+
+[[hql-custom-function-postgresql-example]]
+.Custom PostgreSQL function
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/PostgreSQLFunctionWhereClauseTest.java[tags=hql-custom-function-postgresql-example]
+----
+====
+
+Let's consider we have persisted the following entity in our database:
+
+[[hql-custom-function-postgresql-entity-example]]
+.Book entity
+====
+[source, JAVA, indent=0]
+----
+include::{sourcedir}/PostgreSQLFunctionWhereClauseTest.java[tags=hql-custom-function-postgresql-entity-example]
+----
+====
+
+[[hql-non-standard-functions-where-clause]]
+==== Non-standardized functions referenced in the WHERE clause
```

Comment:
It's a detail but we should use either `custom`, `non-standardized` or `user-defined`. Otherwise it's not clear you're talking about the same thing all along.

The anchors should be consistent as well.

Reply:
User-defined is the best one, I guess. I could pick this one for all occurrences.